components/BottomDrawer.tsx

```diff
@@ -43,6 +44,10 @@ const BottomDrawer = forwardRef<BottomDrawerRef, DrawerProps>((props, ref) => {
     refetchIntervalInBackground: false,
   })
 
+  const handleSheetChanges = useCallback((index: number) => {
+    setIsBottomSheetOpen(index === 1)
+  }, [])
+
```

Comment:
do we need `useCallback ` here? 
```suggestion
  const handleSheetChange = (index: number) => {
    setIsBottomSheetOpen(index === 1)
  }

```

Reply:
also how about renaming the function to "handleSheetChange"

Reply:
About the useCallback, we do need it because `react native bottom sheet` memoizes the props to keep the optimization, so we should pass the memorized  methods as well. 

The naming conveys what the function is doing. reference: https://gorhom.dev/react-native-bottom-sheet/components/bottomsheetbackdrop#pressbehavior
 what do you suggest ?

Reply:
since it's the library's official documentation we can keep it as is